server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -177,88 +122,50 @@ protected void setupFieldType(BuilderContext context) {
             if (name().isEmpty()) {
                 throw new IllegalArgumentException("name cannot be empty string");
             }
-
-            AbstractGeometryFieldType ft = (AbstractGeometryFieldType)fieldType();
-            ft.setOrientation(orientation().value());
         }
     }
 
-    protected static final String DEPRECATED_PARAMETERS_KEY = "deprecated_parameters";
-
     public abstract static class TypeParser implements Mapper.TypeParser {
         protected abstract Builder newBuilder(String name, Map<String, Object> params);
 
-        protected boolean parseXContentParameters(String name, Map.Entry<String, Object> entry, Map<String, Object> params)
-                throws MapperParsingException {
-            if (DeprecatedParameters.parse(name, entry.getKey(), entry.getValue(),
-                (DeprecatedParameters)params.get(DEPRECATED_PARAMETERS_KEY))) {
-                return true;
-            }
-            return false;
-        }
-
-        @Override
-        public Mapper.Builder parse(String name, Map<String, Object> node, ParserContext parserContext) throws MapperParsingException {
-            Map<String, Object> params = new HashMap<>();
-            boolean parsedDeprecatedParameters = false;
-            params.put(DEPRECATED_PARAMETERS_KEY, new DeprecatedParameters());
+        public Builder parse(String name, Map<String, Object> node, Map<String, Object> params, ParserContext parserContext) {
```

Comment:
Ca we use generics here? I think using them will avoid the casting we are doing in the subclasses and it would look cleaner? Something like:

```
public abstract static class TypeParser<T extends Builder> implements Mapper.TypeParser {
```

Reply:
:+1: